src/test/java/testapp/adapter/repository/PersistentUserRepositoryTest.java

```diff
@@ -0,0 +1,88 @@
+package testapp.adapter.repository;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalMemcacheServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import testapp.adapter.listeners.UserLogListener;
+import testapp.core.User;
+import testapp.core.UserRepository;
+
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static testapp.matchers.CustomMatchers.containsUser;
+import static testapp.matchers.CustomMatchers.sameAs;
+
+public class PersistentUserRepositoryTest {
+    private LocalServiceTestHelper helper = new LocalServiceTestHelper(
+            new LocalMemcacheServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() throws Exception {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        helper.tearDown();
+    }
+
+    @Test
+    public void userRegistration() throws Exception {
+        UserRepository repository = new PersistentUserRepository(new UserLogListener());
```

Comment:
the constructor require a listener thats why I put that listener, I think I can replace it with Fake one if you want